scripts/run_acceptance_tests_test.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: UTF-8 -*-
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/run_acceptance_tests.py."""
+
+from __future__ import annotations
+
+import contextlib
+import subprocess
+import sys
+
+from core.constants import constants
+from core.tests import test_utils
+from scripts import build
+from scripts import common
+from scripts import run_acceptance_tests
+from scripts import scripts_test_utils
+from scripts import servers
+
+from typing import ContextManager, Optional
+
+
+def mock_managed_long_lived_process(
+    *unused_args: str, **unused_kwargs: str
+) -> ContextManager[scripts_test_utils.PopenStub]:
+    """Mock method for replacing the managed_process() functions to simulate a
+    long-lived process. This process stays alive for 10 poll() calls, and
+    then terminates thereafter.
+
+    Returns:
+        Context manager. A context manager that always yields a mock
+        process.
+    """
+    stub = scripts_test_utils.PopenStub(alive=True)
+
+    def mock_poll(stub: scripts_test_utils.PopenStub) -> Optional[int]:
+        stub.poll_count += 1
+        if stub.poll_count >= 10:
+            stub.alive = False
+        return None if stub.alive else stub._return_code    # pylint: disable=protected-access
```

Comment:
You can avoid the protected access like this:

```suggestion
        return None if stub.alive else stub.returncode
```